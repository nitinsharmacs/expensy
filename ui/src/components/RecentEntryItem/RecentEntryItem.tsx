import { ListItemButton, ListItemText } from '@mui/material';

export type RecentEntryItemProps = {
  category: string;
  amount: string;
  date: string;
  description: string;
  id: number;
  highlight: boolean;
  onClick: (id: number) => void;
};

type PrimaryContentProps = {
  increment: boolean;
  amount: string;
  date: string;
  category: string;
};

const PrimaryContent = (props: PrimaryContentProps) => (
  <div
    style={{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-between',
    }}
  >
    <span
      style={{ color: props.increment ? 'green' : 'red', fontSize: '26px' }}
    >
      {props.amount}
    </span>
    <span>{props.category}</span>
    <span>{props.date}</span>
  </div>
);

const RecentEntryItem = (props: RecentEntryItemProps) => {
  return (
    <ListItemButton
      sx={{
        borderBottom: '1px solid #dfe1e3',
        '&.Mui-selected': {
          backgroundColor: '#dfe1e3',
        },
      }}
      onClick={() => props.onClick(props.id)}
      selected={props.highlight}
    >
      <ListItemText
        primary={
          <PrimaryContent {...props} increment={!props.amount.includes('-')} />
        }
        secondary={props.description}
      />
    </ListItemButton>
  );
};

export default RecentEntryItem;
