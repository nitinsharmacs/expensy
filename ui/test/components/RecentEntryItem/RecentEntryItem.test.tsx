import * as React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, test, expect } from 'vitest';

import RecentEntryItem from '@components/RecentEntryItem/RecentEntryItem';
import RecentEntryItemProps from '@components/RecentEntryItem/RecentEntryItemProps';

describe('RecentEntryItem', () => {
  
  test('should match the snapshot', () => {
    const props: RecentEntryItemProps = {
      category: 'health',
      amount: '-12',
      date: 'date',
      description: 'string',
      id: 1,
      highlight: false,
      onClick: () => {},
    };
    render(<RecentEntryItem {...props} />);
    expect(screen.getAllByRole('button')).toMatchSnapshot();
  });

  test('should item be green for amount credited', () => {
    const props: RecentEntryItemProps = {
      category: 'health',
      amount: '12',
      date: 'date',
      description: 'string',
      id: 1,
      highlight: false,
      onClick: () => {},
    };

    render(<RecentEntryItem {...props} />);
    const element = screen.getByText('12')
    const style = window.getComputedStyle(element)
    expect(style.color).toBe('rgb(0, 128, 0)')
  });

  test('should item be red for amount debited', () => {
    const props: RecentEntryItemProps = {
      category: 'health',
      amount: '-12',
      date: 'date',
      description: 'string',
      id: 1,
      highlight: false,
      onClick: () => {},
    };

    render(<RecentEntryItem {...props} />);
    const element = screen.getByText('-12')
    const style = window.getComputedStyle(element)
    expect(style.color).toBe('rgb(255, 0, 0)')
  });
});
