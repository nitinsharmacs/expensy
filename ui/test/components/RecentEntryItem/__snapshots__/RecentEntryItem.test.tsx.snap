// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RecentEntryItem > should match the snapshot 1`] = `
[
  <div
    class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1nmdcmv-MuiButtonBase-root-MuiListItemButton-root"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        <div
          style="display: flex; align-items: center; justify-content: space-between;"
        >
          <span
            style="color: red; font-size: 26px;"
          >
            -12
          </span>
          <span>
            health
          </span>
          <span>
            date
          </span>
        </div>
      </span>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
      >
        string
      </p>
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </div>,
]
`;
